modules/lang-mustache/src/main/java/org/elasticsearch/script/mustache/MustachePlugin.java

```diff
@@ -19,15 +19,40 @@
 
 package org.elasticsearch.script.mustache;
 
+import org.elasticsearch.action.ActionModule;
+import org.elasticsearch.action.search.template.MultiSearchTemplateAction;
+import org.elasticsearch.action.search.template.RenderSearchTemplateAction;
+import org.elasticsearch.action.search.template.SearchTemplateAction;
+import org.elasticsearch.action.search.template.TransportMultiSearchTemplateAction;
+import org.elasticsearch.action.search.template.TransportRenderSearchTemplateAction;
+import org.elasticsearch.action.search.template.TransportSearchTemplateAction;
+import org.elasticsearch.client.Client;
+import org.elasticsearch.client.transport.TransportClient;
+import org.elasticsearch.common.network.NetworkModule;
+import org.elasticsearch.common.settings.Settings;
 import org.elasticsearch.plugins.Plugin;
+import org.elasticsearch.rest.action.search.template.RestDeleteSearchTemplateAction;
+import org.elasticsearch.rest.action.search.template.RestGetSearchTemplateAction;
+import org.elasticsearch.rest.action.search.template.RestMultiSearchTemplateAction;
+import org.elasticsearch.rest.action.search.template.RestPutSearchTemplateAction;
+import org.elasticsearch.rest.action.search.template.RestRenderSearchTemplateAction;
+import org.elasticsearch.rest.action.search.template.RestSearchTemplateAction;
 import org.elasticsearch.script.ScriptEngineRegistry;
 import org.elasticsearch.script.ScriptModule;
 
 public class MustachePlugin extends Plugin {
 
+    public static final String NAME = "lang-mustache";
+
+    private final boolean transportClientMode;
+
+    public MustachePlugin(Settings settings) {
+        this.transportClientMode = transportClientMode(settings);
```

Comment:
Instead of this maybe just make `NetworkModule.isTransportClient()` or something? I think that is already available there and I think the setting only matters of that module. Right?